dev_tools/utils/nxdl_utils.py

```diff
@@ -0,0 +1,839 @@
+# pylint: disable=too-many-lines
+"""Parse NeXus definition files
+"""
+
+import os
+import textwrap
+from functools import lru_cache
+from glob import glob
+from pathlib import Path
+
+import lxml.etree as ET
+from lxml.etree import ParseError as xmlER
+
+
+def remove_namespace_from_tag(tag):
+    """Helper function to remove the namespace from an XML tag."""
+
+    return tag.split("}")[-1]
+
+
+class NxdlAttributeNotFoundError(Exception):
+    """An exception to throw when an Nxdl attribute is not found."""
+
+
+def get_nexus_definitions_path():
+    """Check NEXUS_DEF_PATH variable.
+    If it is empty, this function is filling it"""
+    try:  # either given by sys env
+        return Path(os.environ["NEXUS_DEF_PATH"])
+    except KeyError:  # or it should be available locally under the dir 'definitions'
+        local_dir = Path(__file__).resolve().parent
+        for _ in range(2):
+            local_dir = local_dir.parent
+        return local_dir
+
+
+nexus_def_path = get_nexus_definitions_path()
+
+
+def get_app_defs_names():
+    """Returns all the AppDef names without their extension: .nxdl.xml"""
+    app_def_path_glob = nexus_def_path / "applications" / "*.nxdl*"
+
+    contrib_def_path_glob = Path(nexus_def_path) / "contributed_definitions" / "*.nxdl*"
+
+    files = sorted(glob(app_def_path_glob))
+    for nexus_file in sorted(contrib_def_path_glob):
+        root = get_xml_root(nexus_file)
+        if root.attrib["category"] == "application":
+            files.append(nexus_file)
+
+    return [Path(file).name[:-9] for file in files] + ["NXroot"]
```

Comment:
`[:-9]` assumes file ends with `.nxdl.xml` but that's not only what the glob can match

Reply:
DONE